src/main/java/amidst/mojangapi/world/export/WorldExporter.java

```diff
@@ -1,49 +1,158 @@
 package amidst.mojangapi.world.export;
 
+import java.awt.Rectangle;
+import java.awt.image.DataBuffer;
+import java.awt.image.DirectColorModel;
+import java.awt.image.SinglePixelPackedSampleModel;
+import java.io.IOException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
 import java.util.function.Consumer;
+import javax.media.jai.JAI;
+
+import org.jaitools.tilecache.DiskMemTileCache;
+import org.jaitools.tiledimage.DiskMemImage;
+
+import com.sun.media.jai.codec.TIFFEncodeParam;
 
 import amidst.documentation.AmidstThread;
 import amidst.documentation.CalledOnlyBy;
-import amidst.documentation.NotThreadSafe;
+import amidst.documentation.ThreadSafe;
+import amidst.fragment.Fragment;
+import amidst.logging.AmidstLogger;
 import amidst.mojangapi.world.World;
+import amidst.mojangapi.world.biome.UnknownBiomeIndexException;
+import amidst.mojangapi.world.coordinates.CoordinatesInWorld;
+import amidst.settings.biomeprofile.BiomeProfileSelection;
 import amidst.threading.WorkerExecutor;
 import amidst.threading.worker.ProgressReporter;
 
-@NotThreadSafe
+@ThreadSafe
 public class WorldExporter {
-	private final WorkerExecutor workerExecutor;
-	@SuppressWarnings("unused")
+	private static final int TILE_SIZE = Fragment.SIZE;
+	private static final int TILES_BETWEEN_FLUSHES = 30;
+	private static final DiskMemTileCache tileCache = new DiskMemTileCache();
+	private static final ThreadPoolExecutor pool = (ThreadPoolExecutor) Executors.newCachedThreadPool(new ThreadFactory() {
+		private int threadNo;
+		
+		@Override
+		public Thread newThread(Runnable r) {
+			Thread t = new Thread(r);
+			t.setName("WorldExporterThread-" + threadNo++);
+			return t;
+		}
+	});
+	private static final WorkerExecutor exporterExecutor = new WorkerExecutor(pool);
```

Comment:
I feel that this shouldn't create it's own executor, but instead use the one provided by `ThreadMaster.getWorkerExecutor()`, by passing it into the constructor.

Is there a specific reason why you allocated a specific thread pool for this?

Reply:
I have this creating it's own executor so we could have a reliable implementation for the static method `isExporterRunning()`. I tried experimenting with seeing if a thread exists by name so we wouldn't have to do this, but I couldn't reliably get if it was running at the time or not. We may be able to extend Thread and have a volatile flag that says if the thread is active or not that is run before and after the thread executes, but that seems kind of janky.

Reply:
Why not use an static `AtomicLong` field to replace the `pool.getActiveCount()` method?  
Something like:
```java
private static AtomicLong ACTIVE_THREADS = new AtomicLong(0);

public void export() throws WorldExportException {
	exporterExecutor.<String> run(() -> {
		ACTIVE_THREADS.incrementAndGet();
		try {
			this.doExport();
		} finally {
			ACTIVE_THREADS.decrementAndGet();
		}
	}, progressListener, this::onFinished);
}
```

Reply:
Just implemented your solution. I don't know why I didn't think of that earlier, but it's a lot simpler.